### PR TITLE
GTJ-17 frontend

### PR DESCRIPTION
**Pull request by kkeszthe**

### CODE REVIEW

File: frontend/src/components/troops/TroopButton.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+};
+const mapDispatchToProps = dispatch => {
+  return {
+    setError: error => {
+      dispatch(setErrorAction(error));
+    },
+  };
+};
+
+export default connect(mapStateToProps, mapDispatchToProps)(TroopButton);
```
Comment (adamgyulavari) [inline frontend/src/components/troops/TroopButton.js:34]: @kkeszthe this component doesnt use anything from the redux store, especially not the whole state

File: frontend/src/components/troops/Troops.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {troopsSummary.amountPerLevel.map((troopSummary, i) => {
+          return <Button key={i} troopSummary={troopSummary}></Button>;
+        })}
+      </div>
+    </div>
+  );
+}
+
+Troops.propTypes = {
+  kingdom: PropTypes.any.isRequired,
```
Comment (adamgyulavari) [inline frontend/src/components/troops/Troops.js:62]: @kkeszthe proptypes and actual props are not matching

File: frontend/src/components/troops/Troops.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+const mapStateToProps = state => {
+  return state;
```
Comment (adamgyulavari) [inline frontend/src/components/troops/Troops.js:66]: we dont need the whole state, just the kingdom and the troops

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: frontend/src/components/troops/Troops.js (hunk 2 above)
Comment (adamgyulavari) [reply frontend/src/components/troops/Troops.js:62]: this is still not okay, you get a kingdom and a troops array I guess

Comment (kkeszthe) [reply frontend/src/components/troops/Troops.js]: @adamgyulavari I hoped I would still get away with it :-D

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**